src/main/scala/blast.scala

```diff
@@ -62,28 +62,293 @@ abstract class Blast(val version: String) extends Bundle { blast =>
+  case object tblastn       extends BlastCommand("blastp");   type tblastn  = tblastn.type
+  case object tblastx       extends BlastCommand("blastp");   type tblastx  = tblastx.type
+  case object makeblastdb   extends BlastCommand("makeblastdb"); type makeblastdb = makeblastdb.type
+
+  type AllBlasts  = blastn :~: blastp :~: blastx :~: tblastn :~: tblastx :~: ∅
+  val allBlasts   = blastn :~: blastp :~: blastx :~: tblastn :~: tblastx :~: ∅
+
+  sealed trait AnyBlastOption {
+
+    type Commands <: AnyTypeSet.Of[AnyBlastCommand]
+    // val commands: Commands
+
+    def toSeq: Seq[String]
+  }
+  abstract class BlastOption[Cmmnds <: AnyTypeSet.Of[AnyBlastCommand]](
+    val commands: Cmmnds,
+    val toSeq: Seq[String]
+  )
+  extends AnyBlastOption { type Commands = Cmmnds }
+
+  case class numThreads(val number: Int)    extends BlastOption(
+    allBlasts,
+    Seq("-num_threads", s"${number}")
+  )
+  case class db(val file: File)             extends BlastOption(
+    allBlasts,
+    Seq("-db", file.getCanonicalPath().toString)
+  )
+  case class query(val file: File)          extends BlastOption(
+    allBlasts,
+    Seq("-query", file.getCanonicalPath().toString)
+  )
+  case class out(val file: File)            extends BlastOption(
+    makeblastdb :~: allBlasts,
+    Seq("-out", file.getCanonicalPath().toString)
+  )
+  case class evalue(val number: Double)     extends BlastOption(
+    allBlasts,
+    Seq("-evalue", number.toString)
+  )
+  case class maxTargetSeqs(val number: Int) extends BlastOption(
+    allBlasts,
+    Seq("-max_target_seqs", s"${number}")
+  )
+  case object showGis                       extends BlastOption(
+    allBlasts,
+    Seq("-show_gis")
+  )
+  sealed abstract class BlastTask[BC <: BlastCommand](bc: BC, toSeq: Seq[String]) extends BlastOption(
+    bc :~: ∅,
+    toSeq
+  )
+
+  case object task {
+
+    // TODO finish writing all tasks here
+    case object megablast                 extends BlastTask( blastAPI.blastn, Seq("-task", "megablast") )
+    case object dcMegablast               extends BlastTask( blastAPI.blastn, Seq("-task", "dc-megablast") )
+    case object blastn                    extends BlastTask( blastAPI.blastn, Seq("-task", "blastn") )
+    case object blastnShort               extends BlastTask( blastAPI.blastn, Seq("-task", "blastn-short") )
+    case object rmblastn                  extends BlastTask( blastAPI.blastn, Seq("-task", "rmblastn") )
+    case object blastp
+    case object blastpFast
+    case object blastpShort
+    case object blastx
+    case object blastxFast
+    case object tblastn
+    case object tblastnFast
+  }
+
+  sealed trait BlastDBInputType
+  object dbInputType {
+
+    case object asn1_bin  extends BlastDBInputType
+    case object asn1_txt  extends BlastDBInputType
+    case object blastdb   extends BlastDBInputType
+    case object fasta     extends BlastDBInputType
```

Comment:
Could be properties with `Raw = File`, then passing it to a command would be more convenient